internal/payloads/reservation_payload.go

```diff
@@ -2,38 +2,85 @@ package payloads
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/go-chi/render"
 )
 
 // ReservationRequest is empty, account comes in HTTP header and
 // provider type in HTTP URL. All other fields are auto-generated.
-type ReservationRequest struct {
-}
 
-type ReservationResponse struct {
-	*models.Reservation
-}
+type ReservationResponsePayload struct {
+	ID int64 `json:"id"`
+
+	// Provider type. Required.
+	Provider int `json:"provider"`
+
+	// Time when reservation was made.
+	CreatedAt time.Time `json:"created_at"`
 
-type NoopReservationRequest ReservationRequest
+	// Textual status of the reservation or error when there was a failure
+	Status string `json:"status"`
 
-type NoopReservationResponse struct {
-	*models.NoopReservation
+	// Time when reservation was finished or nil when it's still processing.
+	FinishedAt *time.Time `json:"finished_at"`
+
+	// Flag indicating success, error or unknown state (NULL). See Status for the actual error.
+	Success *bool `json:"success"`
 }
 
-type AWSReservationRequest struct {
-	*models.AWSReservation
+type AWSReservationResponsePayload struct {
+	Reservation ReservationResponsePayload
+
+	// Pubkey ID.
+	PubkeyID int64 `json:"pubkey_id"`
+
+	// Source ID.
+	SourceID int64 `json:"source_id"`
+
+	//AWS Instance type.
+	InstanceType string `json:"instance_type"`
+
+	// Amount of instances to provision of type: Instance type.
+	Amount int32 `json:"amount"`
+
+	// The ID of the image from which the instance is created.
+	ImageID string `json:"image_id"`
+
+	// The ID of the aws reservation which was created.
+	AWSReservationID string `json:"aws_reservation_id"`
 }
 
-type AWSReservationResponse struct {
-	*models.AWSReservation
+type NoopReservationResponsePayload struct {
+	ReservationResponsePayload
 }
 
-func (p *ReservationRequest) Bind(_ *http.Request) error {
-	return nil
+type AWSReservationRequestPayload struct {
+	// Pubkey ID.
+	PubkeyID int64 `json:"pubkey_id"`
+
+	// Source ID.
+	SourceID int64 `json:"source_id"`
+
+	//AWS Instance type.
+	InstanceType string `json:"instance_type"`
+
+	// Amount of instances to provision of type: Instance type.
+	Amount int32 ` json:"amount"`
+
+	// The ID of the image from which the instance is created.
+	ImageID string `json:"image_id"`
 }
 
+type AWSReservationRequest AWSReservationRequestPayload
+
+type NoopReservationResponse NoopReservationResponsePayload
+
+type AWSReservationResponse AWSReservationResponsePayload
+
+type ReservationResponse ReservationResponsePayload
```

Comment:
I do not understand these aliases, we simply need the following types:

* AWSReservationRequestPayload
* AWSReservationResponsePayload
* NoopReservationRequestPayload
* NoopReservationResponsetPayload
* GenericReservationResponsePayload (was: ReservationResponsePayload)

Nothing else, no aliases or embedding is needed, these are all unique payloads.